bootstrap/handlers/secret/client/vaultclient_test.go

```diff
@@ -0,0 +1,381 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"reflect"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/interfaces"
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/logging"
+	"github.com/edgexfoundry/go-mod-bootstrap/config"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+
+	"github.com/edgexfoundry/go-mod-secrets/pkg/providers/vault"
+)
+
+type configurationStruct struct {
+	writable    writableInfo
+	logging     config.LoggingInfo
+	secretStore config.SecretStoreInfo
+}
+
+type writableInfo struct {
+	logLevel string
+}
+
+func (c *configurationStruct) UpdateFromRaw(rawConfig interface{}) bool {
+	return true
+}
+
+func (c *configurationStruct) EmptyWritablePtr() interface{} {
+	return &writableInfo{}
+}
+
+func (c *configurationStruct) UpdateWritableFromRaw(rawWritable interface{}) bool {
+	return true
+}
+
+func (c *configurationStruct) GetBootstrap() interfaces.BootstrapConfiguration {
+	// temporary until we can make backwards-breaking configuration.toml change
+	return interfaces.BootstrapConfiguration{
+		Logging:     c.logging,
+		SecretStore: c.secretStore,
+	}
+}
+
+func (c *configurationStruct) GetLogLevel() string {
+	return c.writable.logLevel
+}
+
+func (c *configurationStruct) GetRegistryInfo() config.RegistryInfo {
+	return config.RegistryInfo{}
+}
+
+func TestGetClient(t *testing.T) {
```

Comment:
There's a lot in this test.  Beyond the simplification created by passing in actual dependencies instead of the DIC into your package's receiver, can you simplify further?

Reply:
Simplified